src/main/java/com/agata/jeeshop/services/CartService.java

```diff
@@ -80,10 +80,8 @@ public String calculateTotalAmount(Long id){
         return String.format("%.2f", totalAmount);
     }
 
-
-
-
-   // public void removeProduct(Long id, Long itemId) {
-     //   cartRepository.getOne(id).getItems().remove(itemId);
-    //}
+   public CartItemDto removeProduct(CartItemDto cartItemDto) {
+        Cart cart = cartRepository.save(cartMapper.fromDto(cartItemDto));
```

Comment:
Tutaj próbujesz zrobić dziwną rzecz - zmienić CartItemDto na CartItem i zapisać do bazy używając CartRepository.

Spróbuj coś takiego:
1. pobierz Cart z bazy danych po id
2. wyjmij listę obiektów CartItems z koszyka, przeiteruj się przez nią, jeżeli id itemy jest zgodne z cartItemId to wtedy wywoła cartItemRepository.delete(cartItem)
3. Jeżeli w danym koszyku nie znajdziesz itemu o wymaganym cartId możesz póki co rzucić wyjątkiem: ```throw new RuntimeException('Cart item with id: " + cartItemId + " not found");```